cmake/BLTMacros.cmake

```diff
@@ -979,3 +979,86 @@ macro(blt_combine_static_libraries)
 
     unset( libLinkLine )
 endmacro(blt_combine_static_libraries)
+
+
+
+##------------------------------------------------------------------------------
+## blt_print_target_properties (TARGET <target> )
+##
+## Prints out all properties of the given target.
+##
+## The required target parameteter must either be a valid cmake target 
+## or was registered via blt_register_library.
+##
+## Output is of the form:
+##     [<target> property] <property>: <value>
+## for each property
+##------------------------------------------------------------------------------
+macro(blt_print_target_properties)
+
+    set(options)
+    set(singleValuedArgs TARGET)
+    set(multiValuedArgs)
+
+    ## parse the arguments to the macro
+    cmake_parse_arguments(arg
+         "${options}" "${singleValuedArgs}" "${multiValuedArgs}" ${ARGN})
+
+    ## check for required arguments
+    if(NOT DEFINED arg_TARGET)
+        message(FATAL_ERROR "TARGET is a required parameter for the blt_print_target_properties macro")
+    endif()
+
+    if(TARGET ${arg_TARGET})
+        ## Solution adapted from https://stackoverflow.com/q/32183975
+        ## Create list of cmake properties
+        set(_property_list)
+        execute_process(COMMAND cmake --help-property-list OUTPUT_VARIABLE _property_list)
+        string(REGEX REPLACE ";" "\\\\;" _property_list "${_property_list}")
+        string(REGEX REPLACE "\n" ";" _property_list "${_property_list}")
+        list(FILTER _property_list EXCLUDE REGEX "^LOCATION$|^LOCATION_|_LOCATION$")
```

Comment:
Looks like ``list(FILTER )`` is only available in ``cmake-3.6+`` and our travis is ``cmake-3.3.1``.
I'll find an alternative...